src/sentry_database.c

```diff
@@ -178,8 +178,23 @@ sentry__process_old_runs(const sentry_options_t *options)
                         session_envelope = sentry__envelope_new();
                     }
                     sentry_session_t *session = sentry__session_from_path(file);
+                    // this is just a heuristic: whenever the session was not
+                    // closed properly, and we do have a crash that happened
+                    // *after* the session was started, we will assume that the
+                    // crash corresponds to the session and flag it as crashed.
+                    // this should only happen when using crashpad, and there
+                    // should normally be only a single unclosed session at a
+                    // time.
                     if (session->status == SENTRY_SESSION_STATUS_OK) {
-                        session->status = SENTRY_SESSION_STATUS_ABNORMAL;
+                        bool was_crash
+                            = last_crash && last_crash > session->started_ms;
+                        if (was_crash) {
+                            session->duration_ms
+                                = last_crash - session->started_ms;
```

Comment:
We should probably set `last_crash = 0` in this branch to avoid marking multiple sessions as crashed.